libs/langgraph/langgraph/pregel/remote.py

```diff
@@ -0,0 +1,458 @@
+from typing import (
+    Any,
+    AsyncIterator,
+    Iterator,
+    Optional,
+    Sequence,
+    Union,
+)
+
+import orjson
+from langchain_core.runnables import Runnable, RunnableConfig
+from langchain_core.runnables.graph import (
+    Edge as DrawableEdge,
+)
+from langchain_core.runnables.graph import (
+    Graph as DrawableGraph,
+)
+from langchain_core.runnables.graph import (
+    Node as DrawableNode,
+)
+from langchain_core.runnables.schema import StandardStreamEvent, StreamEvent
+from langgraph_sdk.client import LangGraphClient, SyncLangGraphClient
+from langgraph_sdk.schema import Checkpoint, ThreadState
+from typing_extensions import Self
+
+from langgraph.checkpoint.base import CheckpointMetadata
+from langgraph.pregel.protocol import PregelProtocol
+from langgraph.pregel.types import All, PregelTask, StateSnapshot, StreamMode
+from langgraph.types import Interrupt
+from langgraph.utils.config import merge_configs
+
+
+class RemotePregel(PregelProtocol, Runnable):
+    def __init__(
+        self,
+        client: LangGraphClient,
+        sync_client: SyncLangGraphClient,
+        graph_id: str,
+        config: Optional[RunnableConfig] = None,
+    ):
+        self.client = client
+        self.sync_client = sync_client
+        self.graph_id = graph_id
+        self.config = config
+
+    def copy(self, update: dict[str, Any]) -> Self:
+        attrs = {**self.__dict__, **update}
+        return self.__class__(**attrs)
+
+    def with_config(
+        self, config: Optional[RunnableConfig] = None, **kwargs: Any
+    ) -> Self:
+        self.config = config
+        return self
+
+    def get_graph(
+        self,
+        config: Optional[RunnableConfig] = None,
+        *,
+        xray: Union[int, bool] = False,
+    ) -> DrawableGraph:
+        graph = self.sync_client.assistants.get_graph(
+            assistant_id=self.graph_id,
+            xray=xray,
+        )
+
+        # construct DrawableNodes
+        nodes = {}
+        for node in graph["nodes"]:
+            node_id = node["id"]
+            nodes[node_id] = DrawableNode(
+                id=node_id,
+                name=node.get("name", node_id),
+                data=node.get("data", {}),
+                metadata=node.get("metadata"),
+            )
+
+        return DrawableGraph(
+            nodes=nodes,
+            edges=[DrawableEdge(**edge) for edge in graph["edges"]],
+        )
+
+    async def aget_graph(
+        self,
+        config: Optional[RunnableConfig] = None,
+        *,
+        xray: Union[int, bool] = False,
+    ) -> DrawableGraph:
+        graph = await self.client.assistants.get_graph(
+            assistant_id=self.graph_id,
+            xray=xray,
+        )
+
+        # construct DrawableNodes
+        nodes = {}
+        for node in graph["nodes"]:
+            node_id = node["id"]
+            nodes[node_id] = DrawableNode(
+                id=node_id,
+                name=node.get("name", node_id),
```

Comment:
why do we pass this node_id default here?

Reply:
I just needed to pass some default value. I can set the default to an empty string. `name` in `DrawableNode` is not `Optional`.